rca/utils/formatters.py

```diff
@@ -0,0 +1,96 @@
+page_type_mapping = {
+    "GuidePage": "GUIDE",
+    "ProjectPage": "PROJECT",
+    "ResearchCentrePage": "RESEARCH CENTRE",
+    "ShortCoursePage": "SHORT COURSE",
+    "ProgrammePage": "PROGRAMME",
+}
+
+
+def related_list_block_slideshow(slides):
+    # This formatter can be used to render out slides when defined on the model
+    # with RelatedPageListBlockPage()
+    # E.G slides = StreamField(StreamBlock([("Page", RelatedPageListBlockPage())]))
+    # The reason being that custom field data OR internal pages can be referenced
+    formated_slides = []
+    for slide in slides:
+        for block in slide.value:
+            if block.block_type == "custom_teaser":
+                formated_slides.append(
+                    {
+                        "value": {
+                            "title": block.value["title"],
+                            "summary": block.value["text"],
+                            "image": block.value["image"],
+                            "link": block.value["link"]["url"],
+                            "type": block.value["meta"],
+                        }
+                    }
+                )
+            elif block.block_type == "page":
+                page_type = None
+                page = block.value.specific
+                if page.__class__.__name__ in page_type_mapping:
```

Comment:
You can probably remove the `if` and `page_type = None` because `page_type = page_type_mapping.get(page.__class__.__name__, None)` will return `None` safely if the page class name isn't in the mapping

Reply:
Good spot, thanks!